File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/AssignStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public class AssignStatement : IStatement
    {
        private string id;
        private Expression expression;

        public AssignStatement(string id, Expression expression)
        {
            this.id = id;
            this.expression = expression;
        }

        public ProgramState Execute(ProgramState state)
        {
            int value = this.expression.Evaluate(state.getSymbolTable());
            state.getSymbolTable().Put(id, value);
            return state;
        }

        public override string ToString()
        {
            return id + " = " + this.expression.ToString();
        }
    }
}

File: Semester3/Advanced Methods of Programming/ExamenMAP/src/model/statements/AssignStatement.java
package model.statements;

import model.ProgramState;
import model.exceptions.DivisionByZeroException;
import model.expressions.Expression;
import myCollections.MyIDictionary;

public class AssignStatement implements IStatement {
    private String name;
    private Expression expression;

    public AssignStatement(String name, Expression expression)
    {
        this.name = name;
        this.expression = expression;
    }

    @Override
    public ProgramState execute(ProgramState state) throws Exception {
        int value = 0;
        MyIDictionary<String, Integer> symbolTable = state.getSymbolTable();
        value = expression.evaluate(symbolTable, state.getHeapTable());
        symbolTable.put(name, value);
        return null;
    }

    @Override
    public String toString()
    {
        return name + " = " + expression.toString();
    }
}
